src/main/java/net/minecraftforge/common/ForgeMod.java

```diff
@@ -172,4 +187,22 @@ public void gatherData(GatherDataEvent event)
             gen.addProvider(new ForgeRecipeProvider(gen));
         }
     }
+
+    public void registerRecipeSerialziers(RegistryEvent.Register<IRecipeSerializer<?>> event)
+    {
+        CraftingHelper.register(AndCondition.Serializer.INSTANCE);
+        CraftingHelper.register(FalseCondition.Serializer.INSTANCE);
+        CraftingHelper.register(ItemExistsCondition.Serializer.INSTANCE);
+        CraftingHelper.register(ModLoadedCondition.Serializer.INSTANCE);
+        CraftingHelper.register(NotCondition.Serializer.INSTANCE);
+        CraftingHelper.register(OrCondition.Serializer.INSTANCE);
+        CraftingHelper.register(TrueCondition.Serializer.INSTANCE);
+
+        CraftingHelper.register(new ResourceLocation("forge", "compound"), IngredientNBT.Serializer.INSTANCE);
```

Comment:
These two look back to front. Is this correct?

Reply:
Yes.. yes they are.. Would of caught in my next round of tests but good eye.